perfkitbenchmarker/providers/aws/aws_emr.py

```diff
@@ -52,32 +52,33 @@ class AwsEMR(spark_service.BaseSparkService):
 
   Attributes:
     cluster_id: Cluster identifier, set in superclass.
-    num_workers: Number of works, set in superclass.
-    machine_type: Machine type to use.
     project: Enclosing project for the cluster.
     cmd_prefix: emr prefix, including region
-    region: region in which cluster is located.
+    network: network to use; set if needed by machine type
+    bucket_to_delete: bucket name to delete when cluster is
+    terminated.
   """
 
   CLOUD = providers.AWS
+  SPARK_SAMPLE_LOCATION = 'file:///usr/lib/spark/lib/spark-examples.jar'
   SERVICE_NAME = 'emr'
 
   def __init__(self, spark_service_spec):
     super(AwsEMR, self).__init__(spark_service_spec)
     # TODO(hildrum) use availability zone when appropriate
-    if self.machine_type is None:
-      self.machine_type = DEFAULT_MACHINE_TYPE
-
+    worker_machine_type = self.spec.worker_group.vm_spec.machine_type
+    leader_machine_type = self.spec.master_group.vm_spec.machine_type
     self.cmd_prefix = util.AWS_PREFIX
 
-    if self.spec.zone:
-      region = util.GetRegionFromZone(FLAGS.zones[0])
+    if self.zone:
+      region = util.GetRegionFromZone(self.zone)
       self.cmd_prefix += ['--region', region]
 
     # Certain machine types require subnets.
     if (self.spec.static_cluster_id is None and
-        self.machine_type[0:2] in NEEDS_SUBNET):
-      self.network = aws_network.AwsNetwork.GetNetwork(self.spec)
+        (worker_machine_type[0:2] in NEEDS_SUBNET or
+         leader_machine_type[0:2] in NEEDS_SUBNET)):
+      self.network = aws_network.AwsNetwork.GetNetwork(self)
```

Comment:
GetNetwork is supposed to take a VM not a Spark cluster - this currently works because spark clusters also have zone attributes. At a minimum can we add some comments documenting this and/or updating the documentation on this function (or call it on a VM)?